### PR TITLE
Add instructions for running e2e tests

### PR DESCRIPTION
**Pull request by laurieontech**
Contributing docs don't currently include instructions for running the e2e tests locally beyond running the entire test suite. This is incomplete and adding this information makes development more efficient/encourages people to add tests and verify them before creating PRs.

### CODE REVIEW

File: docs/contributing/setting-up-your-local-dev-environment.md [hunk 1]
```diff
@@ -76,6 +76,12 @@ git checkout package.json; yarn --force
   - To run tests for a single package you can run: `yarn jest <package-name>`.
   - To run a single test file you can run: `yarn jest <file-path>`.
 
+If you're adding e2e tests and want to run them against local changes:
+
+- In the root of the mono repo run `yarn lerna run build --scope=<package-name>` where `package-name` is the changes you're testing
```
Comment (AishaBlake) [inline docs/contributing/setting-up-your-local-dev-environment.md:81]:
```suggestion
- In the root of the monorepo, run `yarn lerna run build --scope=<package-name>` where `package-name` is the directory containing the changes you're testing.
```

I found "is the changes you're testing" a little bit awkward. Feel free to fiddle further! 💖

File: docs/contributing/setting-up-your-local-dev-environment.md [hunk 2]
```diff
@@ -76,6 +76,12 @@ git checkout package.json; yarn --force
...
+- Run `gatsby-dev` inside your specific e2e test directory, for example `e2e-tests/themes/development-runtime`
```
Comment (AishaBlake) [inline docs/contributing/setting-up-your-local-dev-environment.md:82]:
```suggestion
- Run `gatsby-dev` inside your specific e2e test directory, for example `e2e-tests/themes/development-runtime`.
```

File: docs/contributing/setting-up-your-local-dev-environment.md [hunk 3]
```diff
@@ -76,6 +76,12 @@ git checkout package.json; yarn --force
...
+- While the previous step is running open a new terminal window and run `yarn test` in that same e2e test directory
```
Comment (AishaBlake) [inline docs/contributing/setting-up-your-local-dev-environment.md:83]:
```suggestion
- While the previous step is running, open a new terminal window and run `yarn test` in that same e2e test directory.
```

You and these commas! 😅

Comment (laurieontech) [reply docs/contributing/setting-up-your-local-dev-environment.md:83]: I hate commas, ok!

**Pull request merged**